File: java-api/src/main/java/com/fagerland/javaapi/repositories/TimeRepository.java
package com.fagerland.javaapi.repositories;

import com.fagerland.javaapi.models.TimeEntry;
import org.springframework.data.repository.CrudRepository;

import java.util.List;

public interface TimeRepository extends CrudRepository<TimeEntry, Long> {
    List<TimeEntry> findAllByUserEntryId(long userId);
    TimeEntry findFirstByUserEntryIdOrderByStartStampDesc(long userId);
}

File: kotlin-maven-api/src/main/kotlin/com/fagerland/brotime/repositories/TimeRepository.kt
package com.fagerland.brotime.repositories

import com.fagerland.brotime.entities.TimeEntity
import org.springframework.data.repository.CrudRepository

interface TimeRepository : CrudRepository<TimeEntity, Long> {
    fun findAllByUserEntityIdOrderByStartTimeDesc(userId: Long?): List<TimeEntity>
    fun findFirstByUserEntityIdOrderByStartTimeDesc(userId: Long?): TimeEntity?
    fun findFirstByIdAndUserEntityId(timeId: Long?, userId: Long?): TimeEntity?
}
